### ISSUE TITLE
Can not import allennlp.data.fields.array_field 

### ISSUE THREAD

**Issue by MHDBST** [allenai/allennlp#5191]
## Checklist

- [x] I have verified that the issue exists against the `main` branch of AllenNLP.
- [x] I have read the relevant section in the [contribution guide](https://github.com/allenai/allennlp/blob/main/CONTRIBUTING.md#bug-fixes-and-new-features) on reporting bugs.
- [x] I have checked the [issues list](https://github.com/allenai/allennlp/issues) for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/allenai/allennlp/pulls) for existing proposed fixes.
- [x] I have checked the [CHANGELOG](https://github.com/allenai/allennlp/blob/main/CHANGELOG.md) and the [commit log](https://github.com/allenai/allennlp/commits/main) to find out if the bug was already fixed in the main branch.
- [x] I have included in the "Description" section below a traceback from any exceptions related to this bug.
- [x] I have included in the "Related issues or possible duplicates" section beloew all related issues and possible duplicate issues (If there are none, check this box anyway).
- [x] I have included in the "Environment" section below the name of the operating system and Python version that I was using when I discovered this bug.
- [x] I have included in the "Environment" section below the output of `pip freeze`.
- [x] I have included in the "Steps to reproduce" section below a minimally reproducible example.

## Description

When I import `from allennlp.data.fields.array_field import ArrayField` the following error occurs:
<details>
<summary><b>Python traceback:</b></summary>
<p>

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/__init__.py", line 1, in <module>
    from allennlp.data.dataset_readers.dataset_reader import DatasetReader
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/dataset_readers/__init__.py", line 10, in <module>
    from allennlp.data.dataset_readers.ccgbank import CcgBankDatasetReader
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/dataset_readers/ccgbank.py", line 9, in <module>
    from allennlp.data.dataset_readers.dataset_reader import DatasetReader
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/dataset_readers/dataset_reader.py", line 8, in <module>
    from allennlp.data.instance import Instance
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/instance.py", line 3, in <module>
    from allennlp.data.fields.field import DataArray, Field
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/fields/__init__.py", line 7, in <module>
    from allennlp.data.fields.array_field import ArrayField
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/fields/array_field.py", line 10, in <module>
    class ArrayField(Field[numpy.ndarray]):
  File "/home/mbastan/context_home/XP_project/scibert/src/allennlp/allennlp/data/fields/array_field.py", line 49, in ArrayField
    @overrides
  File "/home/mbastan/context_home/anaconda2/envs/mypython3/lib/python3.6/site-packages/overrides/overrides.py", line 88, in overrides
    return _overrides(method, check_signature, check_at_runtime)
  File "/home/mbastan/context_home/anaconda2/envs/mypython3/lib/python3.6/site-packages/overrides/overrides.py", line 114, in _overrides
    _validate_method(method, super_class, check_signature)
  File "/home/mbastan/context_home/anaconda2/envs/mypython3/lib/python3.6/site-packages/overrides/overrides.py", line 135, in _validate_method
    ensure_signature_is_compatible(super_method, method, is_static)
  File "/home/mbastan/context_home/anaconda2/envs/mypython3/lib/python3.6/site-packages/overrides/signature.py", line 93, in ensure_signature_is_compatible
    ensure_return_type_compatibility(super_type_hints, sub_type_hints, method_name)
  File "/home/mbastan/context_home/anaconda2/envs/mypython3/lib/python3.6/site-packages/overrides/signature.py", line 288, in ensure_return_type_compatibility
    f"{method_name}: return type `{sub_return}` is not a `{super_return}`."
TypeError: ArrayField.empty_field: return type `None` is not a `allennlp.data.fields.field.Field`.
```

</p>
</details>

## Related issues or possible duplicates

- None

## Environment

OS: Linux

Python version: 3.6.10

<details>
<summary><b>Output of <code>pip freeze</code>:</b></summary>
<p>

```
absl-py==0.12.0
aiohttp==3.7.4.post0
alabaster==0.7.12
-e git://github.com/ibeltagy/allennlp@ac2b21da6008d0e41d31192ea596153988c000a4#egg=allennlp
antlr4-python3-runtime==4.9.2
argon2-cffi==20.1.0
astor==0.8.1
astroid==2.5.3
astunparse==1.6.3
async-generator==1.10
async-timeout==3.0.1
attrs==20.3.0
Babel==2.9.1
backcall==0.2.0
bleach==3.3.0
BLEURT @ file:///home/mbastan/context_home/XP_project/bleurt
blis==0.2.4
boto3==1.17.52
botocore==1.20.52
cached-property==1.5.2
cachetools==4.2.1
certifi==2020.12.5
cffi==1.14.5
chardet==4.0.0
click==7.1.2
conllu==1.3.1
coverage==5.5
cycler==0.10.0
cymem==2.0.5
Cython==0.29.23
dataclasses==0.8
debugpy==1.3.0
decorator==5.0.7
deepspeed==0.3.14
defusedxml==0.7.1
docutils==0.17
editdistance==0.5.3
entrypoints==0.3
fairseq==0.10.2
fastBPE==0.1.0
filelock==3.0.12
fire==0.4.0
flaky==3.7.0
Flask==1.1.2
Flask-Cors==3.0.10
flatbuffers==1.12
fsspec==0.9.0
ftfy==6.0.1
future==0.18.2
gast==0.3.3
get==2019.4.13
gevent==21.1.2
gitdb==4.0.7
GitPython==3.1.14
google-auth==1.28.1
google-auth-oauthlib==0.4.4
google-pasta==0.2.0
greenlet==1.1.0
grpcio==1.32.0
gsub==0.1.5
h5py==2.10.0
hydra-core==1.0.6
idna==2.10
idna-ssl==1.1.0
imagesize==1.2.0
importlib-metadata==3.10.1
importlib-resources==5.1.2
iniconfig==1.1.1
ipykernel==5.5.3
ipython==7.16.1
ipython-genutils==0.2.0
ipywidgets==7.6.3
isort==5.8.0
itsdangerous==1.1.0
jedi==0.17.2
Jinja2==2.11.3
jmespath==0.10.0
joblib==1.0.1
jsonlines==2.0.0
jsonnet==0.17.0
jsonpickle==2.0.0
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==6.1.12
jupyter-console==6.4.0
jupyter-core==4.7.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.0
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.2
kiwisolver==1.3.1
lazy-object-proxy==1.6.0
lxml==4.6.3
Markdown==3.3.4
MarkupSafe==1.1.1
matplotlib==3.3.4
mccabe==0.6.1
mistune==0.8.4
mock==4.0.3
multidict==5.1.0
murmurhash==1.0.5
mytorch==0.1.0.dev0
nbclient==0.5.3
nbconvert==6.0.7
nbformat==5.1.3
nest-asyncio==1.5.1
ninja==1.10.0.post2
nltk==3.6.1
notebook==6.3.0
numpy==1.19.5
numpydoc==1.1.0
oauthlib==3.1.0
omegaconf==2.0.6
opt-einsum==3.3.0
overrides==6.1.0
packaging==20.9
pandas==1.1.5
pandocfilters==1.4.3
parsimonious==0.8.1
parso==0.7.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==8.2.0
plac==0.9.6
pluggy==0.13.1
portalocker==2.3.0
post==2019.4.13
preshed==2.0.1
prometheus-client==0.10.1
prompt-toolkit==3.0.18
protobuf==3.15.8
psutil==5.8.0
ptyprocess==0.7.0
public==2020.12.3
pubmed-parser @ git+git://github.com/titipata/pubmed_parser.git@ee0fce583cf111b71430660e743fd0d72acd68ed
py==1.10.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.20
pycurl==7.43.0.6
Pygments==2.8.1
pylint==2.7.4
pyparsing==2.4.7
pyrsistent==0.17.3
pytest==6.2.3
pytest-cov==2.11.1
python-dateutil==2.8.1
pytorch-lightning==1.2.8
pytorch-pretrained-bert==0.6.2
pytorch-transformers==1.1.0
pytz==2021.1
PyYAML==5.4.1
pyzmq==22.0.3
qtconsole==5.0.3
QtPy==1.9.0
query-string==2020.12.3
regex==2021.4.4
request==2019.4.13
requests==2.25.1
requests-oauthlib==1.3.0
responses==0.13.3
rouge==1.0.0
rouge-score==0.0.4
rsa==4.7.2
s3transfer==0.3.7
sacrebleu==1.5.1
sacremoses==0.0.44
scikit-learn==0.24.1
scipy==1.5.4
Send2Trash==1.5.0
sentencepiece==0.1.95
six==1.15.0
sklearn==0.0
smmap==4.0.0
snowballstemmer==2.1.0
spacy==2.1.9
Sphinx==4.0.0
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
sqlparse==0.4.1
srsly==1.0.5
style==1.1.6
tensorboard==2.4.1
tensorboard-plugin-wit==1.8.0
tensorboardX==2.2
tensorflow==2.4.1
tensorflow-estimator==2.4.0
tensorflow-gpu==2.4.1
termcolor==1.1.0
terminado==0.9.4
testpath==0.4.4
tf-slim==1.1.0
thinc==7.0.8
threadpoolctl==2.1.0
tokenizers==0.10.2
toml==0.10.2
toposort==1.6
torch==1.8.1
torchmetrics==0.2.0
torchvision==0.9.1
tornado==6.1
tqdm==4.60.0
traitlets==4.3.3
transformers==4.5.1
typed-ast==1.4.3
typing-extensions==3.7.4.3
typing-utils==0.0.3
Unidecode==1.2.0
update==0.0.1
urllib3==1.26.4
wasabi==0.8.2
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==1.0.1
widgetsnbextension==3.5.1
word2number==1.1
wrapt==1.12.1
yarl==1.6.3
zipp==3.4.1
zope.event==4.5.0
zope.interface==5.4.0
```

</p>
</details>

## Steps to reproduce

<details>
<summary><b>Example source:</b></summary>
<p>

```
from allennlp.data.fields.array_field import ArrayField
```

</p>
</details>

**Comment by DreamerDeo** [comment]
+1

**Comment by david-waterworth** [comment]
As a workaround

`pip install overrides==3.1.0`

See https://github.com/allenai/allennlp/issues/5197

**Comment by epwalsh** [comment]
It's interesting that this is popping up now. Are you all using some other package that pins `overrides` to this newer version?

**Comment by MHDBST** [comment]
@epwalsh No, actually there are some version mismatches between the codes. I had to downgrade `allennlp` to one of the previous version to be able to run `train_allennlp_local.sh`

Needless to mention here, but when I install `allennlp` using the instructions [here](https://github.com/allenai/allennlp) via Conda, after running `train_allennlp_local.sh` I get this error:

`TypeError: '_NamespacePath' object is not subscriptable`

after searching for a while I was able to solve the problem by downgrading torch, allnenlp, and some other dependencies. But the code is still not working. Although I used the solution @david-waterworth kindly suggested but some other issues came up and still I can not train and get the same results as proposed in the paper.
